3696/3/app/models/user.rb

```diff
@@ -0,0 +1,16 @@
+class User < ActiveRecord::Base
+  include BCrypt
+
+  validates_presence_of :name, :email, :password_hash
+  validates_uniqueness_of :name, :email
+  has_many :reviews, dependent: :destroy
+
+  def password
+    @password ||= Password.new(password_hash)
+  end
+
+  def password=(new_password)
+    @password = Password.create(new_password)
```

Comment:
I looked through sources couple of times. what is `Password`? Where is it defined?

Reply:
In bcrypt gem

Reply:
# Initializes a BCrypt::Password instance with the data from a stored hash.
    def initialize(raw_hash)
      if valid_hash?(raw_hash)
        self.replace(raw_hash)
        @version, @cost, @salt, @checksum = split_hash(self)
      else
        raise Errors::InvalidHash.new("invalid hash")
      end
    end

Reply:
def create(secret, options = {})
        cost = options[:cost] || BCrypt::Engine.cost
        raise ArgumentError if cost > 31
        Password.new(BCrypt::Engine.hash_secret(secret, BCrypt::Engine.generate_salt(cost)))
      end

Reply:
I would suggest then not to include BCrypt, but use directly `BCrypt::Password`, so nobody will be confused about its nature.